src/ContainerAppHelper.ts

```diff
@@ -0,0 +1,434 @@
+     * Determines if the provided resource group exists.
+     * @param resourceGroup - the name of the resource group
+     * @returns true if the resource group exists, false otherwise
+     */
+    public async doesResourceGroupExist(resourceGroup: string): Promise<boolean> {
+        toolHelper.writeDebug(`Attempting to determine if resource group "${resourceGroup}" exists`);
+        try {
+            let command = `az group show -n ${resourceGroup} -o none`;
+            let executionResult = await util.executeAndThrowIfError(command);
+            return executionResult.exitCode === 0;
+        } catch (err) {
+            toolHelper.writeWarning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Gets the default location for the Container App provider.
+     * @returns the default location if found, otherwise 'eastus2'
+     */
+    public async getDefaultContainerAppLocation(): Promise<string> {
+        toolHelper.writeDebug(`Attempting to get the default location for the Container App service for the subscription.`);
+        try {
+            let command = `az provider show -n Microsoft.App --query \"resourceTypes[?resourceType=='containerApps'].locations[] | [0]\"`
+            let executionResult = await util.executeAndThrowIfError(command);
+            // If successful, strip out double quotes, spaces and parentheses from the first location returned
+            return executionResult.exitCode === 0 ? executionResult.stdout.toLowerCase().replace(/["() ]/g, "").trim() : `eastus2`;
+        } catch (err) {
+            toolHelper.writeWarning(err.message);
+            return `eastus2`;
+        }
+    }
+
+    /**
+     * Creates a new resource group in the provided location.
+     * @param name - the name of the resource group to create
+     * @param location - the location to create the resource group in
+     */
+    public async createResourceGroup(name: string, location: string) {
+        toolHelper.writeDebug(`Attempting to create resource group "${name}" in location "${location}"`);
+        try {
+            let command = `az group create -n ${name} -l ${location}`;
+            await util.executeAndThrowIfError(command);
+        } catch (err) {
+            toolHelper.writeError(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Gets the name of an existing Container App Environment in the provided resource group.
+     * @param resourceGroup - the resource group to check for an existing Container App Environment
+     * @returns the name of the existing Container App Environment, null if none exists
+     */
+    public async getExistingContainerAppEnvironment(resourceGroup: string) {
+        toolHelper.writeDebug(`Attempting to get the existing Container App Environment in resource group "${resourceGroup}"`);
+        try {
+            let executionResult = await util.executeAndThrowIfError(`az containerapp env list -g ${resourceGroup} --query [0].name`);
```

Comment:
nit: can we move the command that we're executing to its own `command` variable?

Reply:
Updated.